sdk/keyvault/Azure.Security.KeyVault.Certificates/README.md

```diff
@@ -0,0 +1,214 @@
+`StartCreateCertificate` creates a Certificate to be stored in the Azure Key Vault. If a certificate with 
+the same name already exists, then a new version of the certificate is created.
+When creating the certificate the user can specify the policy which controls the certificate lifetime. If no policy is speicired the default policy will be used. The `StartCreateCertificate` operation returns a `CertificateOperation`. The following example creates a self signed certificate with the default policy.
+```c#
+// create a certificate, this starts a long running operation to create and sign the certificate
+CertificateOperation operation = await Client.StartCreateCertificateAsync("MyCertificate");
+
+// optionally you can await the completion of the certificate opteration
+// NOTE: 
+CertificateWithPolicy certificate = await WaitForCompletion(operation);
+```
+
+> NOTE: Depending on the certificate issuer and validation methods, certificate creation and signing can take an indeterministic amount of time. Users should only wait on certificate operations when the operation can be reasonably completed in the scope of the application, such as with self signed certificates or issuers with well known response times.
+
+### Retrieve a Certificate
+`GetCertificateWithPolicy` retrieves the latest version of a certificate stored in the Key Vault along with its `CertificatePolicy`.
+```c#
+CertificateWithPolicy certificateWithPolicy = await Client.GetCertificateWithPolicyAsync("MyCertificate");
+```
+
+`GetCertificate` retrieves a specific version of a certificate in the vault.
+```c#
+Certificate certificate = await Client.GetCertificateAsync(certificateWithPolicy.Name, certificateWithPolicy.Version);
+```
+### Update an existing Certificate
+`UpdateCertificate` updates a certificate stored in the Key Vault.
+```c#
+IDictionary<string, string> tags = new Dictionary<string, string>() { { "key1", "value1" } };
+
+Certificate updated = await Client.UpdateCertificateAsync(certName, certVersion, tags: expTags);
+```
+
+### Delete a Certificate
+`delete_certificate` deletes all versions of a certificate stored in the Key Vault. When [soft-delete][soft_delete] 
+is not enabled for the Key Vault, this operation permanently deletes the certificate. If soft delete is enabled the certificate is marked for deletion and can be optionally purged or recovered up until its scheduled purge date.
+```c#
+DeletedCertificate deletedCert = await Client.DeleteCertificateAsync(certName);
+
+Console.WriteLine(deletedCert.ScheduledPurgeDate);
+
+await Client.PurgeDeletedCertificate(certName);
+```
+
+## Troubleshooting
+
+### General
+When you interact with the Azure Key Vault Key client library using the .NET SDK, errors returned by the service correspond to the same HTTP status codes returned for [REST API][keyvault_rest] requests.
+
+For example, if you try to retrieve a Key that doesn't exist in your Key Vault, a `404` error is returned, indicating `Not Found`.
```

Comment:
"Certificate". I'll not repeat this comment (may be more). I'm seeing a lot of (copy and paste?) references to Key Client and Keys.